### ISSUE TITLE
catkin_find does not respect --first-only option

### ISSUE THREAD

**Issue by tfoote** [ros/catkin#204]
```
$ catkin_find rospack package.xml --first-only
Could not find unique path, the following paths are matching:
/tmp/minimal/rospack/package.xml
/opt/ros/groovy/share/rospack/package.xml
```

```

tfoote@btn:/tmp/minimal/catkin$ catkin_find rospack --share  --first-only
Could not find unique path, the following paths are matching:
/tmp/minimal/build/buildspace/share/rospack
/tmp/minimal/rospack
/opt/ros/groovy/share/rospack
```

**Comment by tkruse** [comment]
I would fix this if I understood the semantics of --first-only
Especially in the case of share, I have no idea which should come first. A similar problem appears when giving no project:
catkin_find --etc --share  --first-only
with projects a and b, and existing folders share/a and etc/b. Which opf those two folders is supposed to be the first?

**Comment by dirk-thomas** [comment]
fixed - latest change on that topic was d328c8c8b0e81596424699d0d2e67a357deb886a
